model/src/pyrenew/arrayutils.py

```diff
@@ -112,138 +112,116 @@ def __repr__(self):
-        period_size : int
-            Size of the period.
-        broadcast_type : str
-            Type of broadcasting to use, either "repeat" or "tile".
-
-        Notes
-        -----
-        See the sample method for more information on the broadcasting types.
-
-        Returns
-        -------
-        None
-        """
-
-        self.validate(
-            offset=offset,
-            period_size=period_size,
-            broadcast_type=broadcast_type,
-        )
+    Parameters
+    ----------
+    data : ArrayLike
+        Data to broadcast.
+    n_timepoints : int
+        Duration of the sequence.
+    offset : int
+        Relative point at which data starts, must be a non-negative integer.
+        Default is zero, i.e., no offset.
+
+    Notes
+    -----
+    Using the `offset` parameter, the function will start the broadcast
+    from the `offset`-th element of the data. If the data is shorter than
+    `n_timepoints`, the function will repeat or tile the data until it
+    reaches `n_timepoints`.
 
-        self.period_size = period_size
-        self.offset = offset
-        self.broadcast_type = broadcast_type
-
-        return None
-
-    @staticmethod
-    def validate(offset: int, period_size: int, broadcast_type: str) -> None:
-        """
-        Validate the input parameters.
-
-        Parameters
-        ----------
-        offset : int
-            Relative point at which data starts, must be between 0 and
-            period_size - 1.
-        period_size : int
-            Size of the period.
-        broadcast_type : str
-            Type of broadcasting to use, either "repeat" or "tile".
-
-        Returns
-        -------
-        None
-        """
-
-        # Period size should be a positive integer
-        assert isinstance(
-            period_size, int
-        ), f"period_size should be an integer. It is {type(period_size)}."
-
-        assert (
-            period_size > 0
-        ), f"period_size should be a positive integer. It is {period_size}."
-
-        # Data starts should be a positive integer
-        assert isinstance(
-            offset, int
-        ), f"offset should be an integer. It is {type(offset)}."
-
-        assert (
-            0 <= offset
-        ), f"offset should be a positive integer. It is {offset}."
-
-        assert offset <= period_size - 1, (
-            "offset should be less than or equal to period_size - 1."
-            f"It is {offset}. It should be less than or equal "
-            f"to {period_size - 1}."
-        )
+    Returns
+    -------
+    ArrayLike
+        Tiled data.
+    """
+
+    # Data starts should be a positive integer
+    assert isinstance(
+        offset, int
+    ), f"offset should be an integer. It is {type(offset)}."
+
+    assert 0 <= offset, f"offset should be a positive integer. It is {offset}."
+
+    return jnp.tile(data, (n_timepoints // data.size) + 1)[
+        offset : (offset + n_timepoints)
+    ]
+
+
+def repeat_until_n(
```

Comment:
I don't think the descriptions of the `offset` parameter are correct here. The offset applies after the tiling, not before.

Reply:
Good catch. Just corrected it.